exercise/static/exercise/chapter.js

```diff
@@ -561,8 +563,12 @@
 					});
 				});
 			} else {
-				chapter.openModal(chapter.messages.submit);
 				var exercise = this;
+				if (exercise.quiz) {
+					exercise.quizSubmission("submit");
+				} else {
+					chapter.openModal(chapter.messages.submit);
```

Comment:
These could include a comment `/* TODO: migrate to overlay */` as you are adding a new branch here (same for the other instances of the same thing too).

Reply:
And commented the branches as well